### PR TITLE
Cosmo: make flrw semi-private API private

### PR DESCRIPTION
**Pull request by nstarman**
Signed-off-by: Nathaniel Starkman (@nstarman) <nstarkman@protonmail.com>

@eerovaher, as discussed, a followup PR to make semi-private API actually private 👍 

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/cosmology/flrw/base.py [hunk 1]
```diff
@@ -31,17 +31,17 @@ def quad(*args, **kwargs):
 
 # Some conversion constants -- useful to compute them once here and reuse in
 # the initialization rather than have every object do them.
-H0units_to_invs = (u.km / (u.s * u.Mpc)).to(1.0 / u.s)
```
[the remark this answers is not in the document]
Comment (nstarman) [reply astropy/cosmology/flrw/base.py:34]: Some of this is quite old. But never in `__all__` nor in the top namespace.

[the remark this answers is not in the document]
Comment (nstarman) [reply astropy/cosmology/flrw/base.py:34]: And I can't find any usage on GH.

**Review by eerovaher — CHANGES_REQUESTED** [review]
None of what is being renamed here is a part of the public API, as the variables are not included in any `__all__` and there is no public documentation for  them either. This pull request simply ensures that the names of what are already private variables follow PEP 8, so I see no reason not to merge those changes. However, because the public API is not being affected I think the one commit should be amended so that there would be no change log entry.

**Comment by nstarman** [comment]
I suppose a way to deprecate and ensure minimal complaints is to put a ``__getattr__`` in ``flrw.__init__`` and raise a deprecation warning. Complete removal happens in ``v6``.

**Comment by pllim** [comment]
Thanks! I think the deprecation would make things less surprising. Need a change log now, perhaps?

**Review by eerovaher — APPROVED** [review]
As far as I can tell the addition of the variables this PR would deprecate was never announced in the change log and they were private to begin with, just with names that did not comply with PEP 8. I still don't see any need for a change log entry.

**Comment by nstarman** [comment]
@pllim I think I agree with @eerovaher on the changelog. You're suggestion to offer a deprecation period is nice and good for users, but these were and are private variables, so I think they don't merit a changelog.

**Comment by nstarman** [comment]
Thanks @eerovaher & @pllim for the reviews and suggestions!

**Pull request merged**